api.go

```diff
@@ -32,9 +31,15 @@ const (
 	API_ERR_INTERNAL
 )
 
+type deploymentsResult struct {
+	deployments []DataDeployment
+	err         error
+}
+
 var (
-	deploymentsChanged = make(chan string)
-	addSubscriber      = make(chan chan string)
+	deploymentsChanged = make(chan interface{})
```

Comment:
the basic concern is that you have a potential blocking method inside a `select - case` loop. So you have a blocking chain. Buffered channel can improve concurrency. You want your `debounce` act as a buffer layer, so sending to a slow thread should not block your receiving from a fast thread and appending.